lib/active_model_serializers/adapter/attributes.rb

```diff
@@ -34,7 +34,7 @@ def serializable_hash_for_single_resource(options)
       def resource_relationships(options)
         relationships = {}
         serializer.associations(@include_tree).each do |association|
-          relationships[association.key] = relationship_value_for(association, options)
+          relationships[association.key] ||= relationship_value_for(association, options)
```

Comment:
That's weird AF. If `serializer.associations(@include_tree).each` yields an association multiple times, it means `self.class._reflections.each` ([here](https://github.com/rails-api/active_model_serializers/blob/master/lib/active_model/serializer/associations.rb#L90)) yields them several times.